homeassistant/components/eq3btsmart/coordinator.py

```diff
@@ -0,0 +1,229 @@
+"""Coordinator for the eq3btsmart integration."""
+
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from eq3btsmart import Thermostat
+from eq3btsmart.exceptions import Eq3Exception
+from eq3btsmart.models import DeviceData, Status
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.device_registry as dr
+from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH, format_mac
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    DEVICE_MODEL,
+    MANUFACTURER,
+    SCAN_INTERVAL,
+    SIGNAL_THERMOSTAT_DISCONNECTED,
+)
+from .models import Eq3Config
+
+_LOGGER = logging.getLogger(__name__)
+
+type Eq3ConfigEntry = ConfigEntry[Eq3ConfigEntryData]
+
+
+@dataclass(slots=True)
+class Eq3ConfigEntryData:
+    """Config entry for a single eQ-3 device."""
+
+    eq3_config: Eq3Config
+    thermostat: Thermostat
+    coordinator: Eq3Coordinator
+
+
+class Eq3Coordinator(DataUpdateCoordinator[Status]):
+    """Coordinator for the eq3btsmart integration."""
+
+    config_entry: Eq3ConfigEntry
+    _mac_address: str
+    _status_future: asyncio.Future[Status] | None
+    _device_data_future: asyncio.Future[DeviceData] | None
+
+    def __init__(
+        self, hass: HomeAssistant, entry: Eq3ConfigEntry, mac_address: str
+    ) -> None:
+        """Initialize the coordinator."""
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=format_mac(mac_address),
+            update_interval=timedelta(seconds=SCAN_INTERVAL),
+            config_entry=entry,
+            always_update=False,
+        )
+
+        self._mac_address = mac_address
+        self._status_future = None
+        self._device_data_future = None
```

Comment:
```suggestion
        self._status_future: asyncio.Future[Status] | None= None
        self._device_data_future: asyncio.Future[DeviceData] | None = None
```
allows us to save some loc